app/subscribers/solidus_stripe/webhook/payment_intent_subscriber.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module SolidusStripe
+  module Webhook
+    # Handlers for Stripe payment_intent events.
+    class PaymentIntentSubscriber
+      include Omnes::Subscriber
+
+      handle :"stripe.payment_intent.succeeded", with: :complete_payment
+
+      # Completes a payment.
+      #
+      # Marks a Solidus payment associated to a Stripe payment intent as
+      # completed.
+      #
+      # @param event [SolidusStripe::Webhook::Event]
+      def complete_payment(event)
+        payment_intent_id = event.data.object.id
+        payment = Spree::Payment.find_by!(response_code: payment_intent_id)
+        payment.complete! unless payment.completed?
```

Comment:
After the event has been processed, how about saving a `SolidusStripe::LogEntries.payment_log` related to the event on the payment?
By doing so, it would enable the admin side to display a timeline of events related to the payment, allowing for a clearer understanding of why, in this case, the payment is marked as completed 🤔 WDYT?

Reply:
Makes total sense. I just added that. Thanks for your suggestion :+1: